Source/Libraries/openXDA.APIAuthentication/APIAuthenticationMiddleware.cs

```diff
@@ -174,6 +206,25 @@ private string GetAPIToken(string registrationKey)
             }
         }
 
+        private bool UseImpersonation(AuthorizationHeader header)
+        {
+            string registrationKey = header.APIKey;
+
+            if (registrationKey is null || header.ImpersonationToken is null)
+                return false;
+
+            using (AdoDataConnection connection = ConnectionFactory())
+            {
+                const string QueryFormat =
+                    "SELECT AllowImpersonation " +
+                    "FROM APIAccessKey " +
+                    "WHERE RegistrationKey = {0} AND (Expires IS NULL OR Expires > GETUTCDATE())" +
+                    "UNION SELECT 0 ORDER BY AllowImpersonation DESC";
```

Comment:
Let's keep this simple. `RegistrationKey` is unique so the `ORDER BY`  shouldn't be necessary. If the query returns no results, it will take the value of `default(bool)` which is `false`.

```suggestion
                const string QueryFormat =
                    "SELECT AllowImpersonation " +
                    "FROM APIAccessKey " +
                    "WHERE RegistrationKey = {0} AND (Expires IS NULL OR Expires > GETUTCDATE())";
```